### ISSUE TITLE
Linuxmint 17.1 Client(2.0.1) Hidden subfolders+files not downloaded even when selected for a fresh ownCloud root folder

### ISSUE THREAD

**Issue by astucieux** [owncloud/client#3876]
Like a few other OC users, I've been caught out by the "client 2.x.x" default action change to ignore hidden files and folders. See:#3688. Personally I think the enhancement in 2.x.x should have been "click to ignore hidden files". Far less disruptive..... Never mind, what's done is done.

The Problem with this Processing Change:
On my domestic ownCloud  8.1.1, a various number of hidden folders are used to store encrypted user data handled by EncFS. On the desktop, if I sign into my OC account with a fresh OC-client session to a NEW OC-root folder these hidden folders are not downloaded. In fact, even if I only selected these DOT prefixed hidden folders, they do NOT get downloaded. Ooops, that can't be right!
Surely I've implied I want to handle hidden stuff because I selected my account hidden folders for syncing. When I go to the OC-client "Edit Ignored Files" and tick the "sync hidden files", nothing will happen. The requested (hidden) DOT-folder(s) never appears in the new OC-root-folder. It will remain empty. The only way to get the hidden folders downloaded by OC is to trigger a sync-run by causing an update in the root structure or updates to a visible folder, if one was present. Another alternative is to bounce the OC-client because when it comes up again the "sync hidden files" is now registering "tick" (true). This work-around is an unfortunate messy hassle.

Please can I suggest the enhancement that the OC account should maintain the "flag" on how to process hidden folders/files, or to have the option available at the sync selection set-up stage. At the very least, if DOT prefixed sub-folders are to be eligible for syncing (downloading) on a fresh OC-root-folder, that the "sync hidden files" setting is automatically "ticked" (set=true) and the correct data downloaded.
(My thanks and gratitude) R.I.P Wuala

**Comment by ckamm** [comment]
@astucieux If I understand you correctly you have two issues:
1) When selecting 'sync hidden files' you have to manually trigger a new sync run to get the data. This should instead happen automatically.
2) It's confusing that you can select dot-prefixed folders in the selective sync interface when the folders will consequently be ignored anyway. They should probably not be shown at all.

**Comment by ckamm** [comment]
I addressed 1) in the patch above.

**Comment by dragotin** [comment]
Yes, I think 2) happens because the selective sync list does not check the ignore patterns.

Thanks for picking that up!

**Comment by astucieux** [comment]
@ckamm It can be described as two issues but, for me, actually solved by one straight forward OC-client adjustment.... If I have DOT prefixed hidden folders under my OC account then I would expect the 'Sync hidden files' to be assumed as 'True' and NOT to have the burden of afterwards manually ticking the flag and then waiting for the downloads.

So to answer your two questions:
1) When selecting 'sync hidden files' you have to manually trigger a new sync run to get the data. This should instead happen automatically.
(CORRECT, but I feel the OC-client should have the logic to see I had already chosen eligible DOT hidden files for syncing. I shouldn't need to tick this flag at a later time to finally get at the hidden folder sync-data.)

2) It's confusing that you can select dot-prefixed folders in the selective sync interface when the folders will consequently be ignored anyway. They should probably not be shown at all.
(PARTIALLY CORRECT) Yes, it is confusing that the OC-client has ignored my account DOT hidden folders, especially when they might be the only ONE I've manually selected. However, Yikes! PLEASE DO NOT INHIBIT the display of these hidden folders at the OC-client sync-selection interface. If you did that I could no longer select specific hidden folders for download and sync processing by the OC-client. (REMEMBER that the OC-client always sets the 'Sync hidden files' OFF (False) when it is creating a NEW ownCloud root folder.)

I can see that your patch #3876 helps handle processing with the status change of the 'Sync hidden files' better. And for me on Linux, hidden folders/files are an important asset due to their regular use. As are (also) symlinks! (Extremely useful but hard to protect)

I'm puzzled as to what or who caused this surprise processing change of hidden folder/files for the OC-client 2.0.x. Was it those folks in "Windows-World"?..... Curse them, if it was!

Basically, if ownCloud was a bit smarter and thereby recognised to handle DOT hidden folders like it did in OC-client 1.8.X, a lot of disruption and disturbance is avoided.

My thanks to you OC guys+girls for pioneering and supporting open-source 'Access', 'Control' and 'Protection' of DATA. Especially as my Wuala account went R/O yesterday (RIP).

**Comment by ckamm** [comment]
@astucieux Excludes are now applied to the selective sync dialogs. That means if hidden files aren't synced these hidden folders won't appear there. Hopefully that'll remove the confusing issue where a folder with a checkmark isn't synced anyway.

**Comment by astucieux** [comment]
@ckamm Greetings..... 
Have I understood your amendment correctly? Will the OC-client sync-selection interface now include the display 'Sync hidden files' option? Meaning, if I tick the flag, my hidden folders will also be displayed and I could then select the one hidden folder I want downloaded to a new OC root-folder location.
If that is how your enhancement will work, that will be brilliant....!

**Comment by ckamm** [comment]
@astucieux If you enable 'sync hidden files' your dot-folders will appear in in the sync-selection interface. If it's disabled, they won't.

**Comment by dragotin** [comment]
@astucieux is your issue resolved? If not, please speak up and we can reopen this. Thanks

**Comment by astucieux** [comment]
@dragotin Yes, thank you! I'm now able to handle my hidden folders and files more easily with this new ownCloud-client "Sync hidden files" feature. I just have to remember to check it is still "ticked" after a new release of the OC client is installed.
